Source/Core/loadImageFromTypedArray.js

```diff
@@ -31,22 +31,29 @@ define([
             type : format
         });
 
-        var blobUrl = window.URL.createObjectURL(blob);
-        var resource = new Resource({
-            url: blobUrl,
-            request: request
-        });
-        return resource.fetchImage({
-            flipY : flipY,
-            preferImageBitmap : true
-        })
-            .then(function(image) {
-                window.URL.revokeObjectURL(blobUrl);
-                return image;
-            })
-            .otherwise(function(error) {
-                window.URL.revokeObjectURL(blobUrl);
-                return when.reject(error);
+        return Resource.supportsImageBitmapOptions()
+            .then(function(result) {
+                if (result) {
+                    return when(Resource.createImageBitmapFromBlob(blob, flipY));
+                }
+
+                var blobUrl = window.URL.createObjectURL(blob);
+                var resource = new Resource({
+                    url: blobUrl,
+                    request: request
+                });
+
+                return resource.fetchImage({
+                    flipY : flipY
+                })
+                .then(function(image) {
```

Comment:
move this `then` out one scope.  ie, prefer
```js
promise
    .then(function(){
        return promise2        
    })
    .then(function(){
        ....
        return result;
    }
```

instead of

```js
promise
    .then(function(){
        return promise2
            .then(function(){
                ....
                return result;
            });
    });
```

Reply:
Thanks for the tip!